### PR TITLE
fixed gce machine types in all region

### PR DESCRIPTION
**Pull request by prekoa**

### CODE REVIEW

File: pkg/productinfo/gce/productinfo_gce.go [hunk 1]
```diff
@@ -221,33 +220,48 @@ func (g *GceInfoer) GetAttributeValues(attribute string) (productinfo.AttrValues
...
+					// minimum 1 Gbps network performance for each virtual machine
+					ntwPerf = 1
+				} else if mt.GuestCpus > 8 {
+					// theoretical maximum of 16 Gbps for each virtual machine
+					ntwPerf = 16
+				} else {
+					// each vCPU has a 2 Gbps egress cap for peak performance
+					ntwPerf = int(mt.GuestCpus * 2)
+				}
+				if len(vms) == 0 {
```
Comment (stoader) [inline pkg/productinfo/gce/productinfo_gce.go]: Consider using a map for tracking the vm types already processed where the key would be the vm type. (`vms map[string]*productinfo.VmInfo`) its more efficient than iterating over and over again on `vms`.

At the end collect the values of the map into a slice and return that slice when exiting the function

File: pkg/productinfo/gce/productinfo_gce.go [hunk 2]
```diff
@@ -226,28 +226,43 @@ func (g *GceInfoer) GetProducts(regionId string) ([]productinfo.VmInfo, error) {
...
-				ntwPerf = strconv.Itoa(16)
-			} else {
-				// each vCPU has a 2 Gbps egress cap for peak performance
-				ntwPerf = strconv.Itoa(int(mt.GuestCpus * 2))
+	for _, zone := range zones {
+		err = g.computeSvc.MachineTypes.List(g.projectId, zone).Pages(context.TODO(), func(allMts *compute.MachineTypeList) error {
+			for _, mt := range allMts.Items {
+				if mt.GuestCpus < 1 {
+					// minimum 1 Gbps network performance for each virtual machine
+					ntwPerf = strconv.Itoa(1)
```
[the remark this answers is not in the document]
Comment (prekoa) [reply pkg/productinfo/gce/productinfo_gce.go]: thx, fixed

File: pkg/productinfo/gce/productinfo_gce.go (hunk 1 above)
Comment (prekoa) [reply pkg/productinfo/gce/productinfo_gce.go]: fixed

File: pkg/productinfo/gce/productinfo_gce.go [hunk 3]
```diff
@@ -220,37 +219,45 @@ func (g *GceInfoer) GetAttributeValues(attribute string) (productinfo.AttrValues
...
-			} else {
-				// each vCPU has a 2 Gbps egress cap for peak performance
-				ntwPerf = strconv.Itoa(int(mt.GuestCpus * 2))
+	for _, zone := range zones {
+		err = g.computeSvc.MachineTypes.List(g.projectId, zone).Pages(context.TODO(), func(allMts *compute.MachineTypeList) error {
+			for _, mt := range allMts.Items {
+				if mt.GuestCpus < 1 {
+					// minimum 1 Gbps network performance for each virtual machine
+					ntwPerf = 1
+				} else if mt.GuestCpus > 8 {
```
[the remark this answers is not in the document]
Comment (prekoa) [reply pkg/productinfo/gce/productinfo_gce.go]: fixed, thx

**Pull request merged**